src/ClusterManager/framework.py

```diff
@@ -807,10 +808,10 @@ def transform_inference_job(params, cluster_config):
 
     roles = {
         "master":
-            Role("master", 1, image, envs, master_resource, 1, 1),
+            Role("master", 1, image, envs, master_resource, -1, -1, -1),
```

Comment:
@leigaoms can you help confirm if this is the desired terminating condition for an inference job. This basically says only when the sole `master` succeeds, the job finishes successfully, otherwise any failure in `master` or `worker` will trigger retry indefinitely.

Reply:
We need to separate if the pod is preempted or it's really failed. Approve this PR first, and I will add another function to judge the exit code of the process.